File: problems/3487-maximum-unique-subarray-sum-after-deletion/solution.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int maxSum(int[] nums) {
        int maxSum = 0;
        int currentSum = 0;
        int left = 0;
        Set<Integer> seen = new HashSet<>();

        for (int right = 0; right < nums.length; right++) {
            while (seen.contains(nums[right])) {
                seen.remove(nums[left]);
                currentSum -= nums[left];
                left++;
            }
            seen.add(nums[right]);
            currentSum += nums[right];
            maxSum = Math.max(maxSum, currentSum);
        }

        return maxSum;
    }
}

File: problems/3487-maximum-unique-subarray-sum-after-deletion/solution.c
int maxSum(int* nums, int numsSize) {
    int seen[201] = {0};
    int currentSum = 0;
    int maxSum = 0;
    int left = 0;

    for (int right = 0; right < numsSize; right++) {
        int num = nums[right];
        num += 100;

        while (seen[num] > 0) {
            int leftNum = nums[left];
            leftNum += 100;
            seen[leftNum]--;
            currentSum -= nums[left];
            left++;
        }

        seen[num]++;
        currentSum += nums[right];
        maxSum = (currentSum > maxSum) ? currentSum : maxSum;
    }

    return maxSum;
}